app/services/user_destroyer.rb

```diff
@@ -32,6 +32,8 @@ def destroy(user, opts = {})
       Draft.where(user_id: user.id).delete_all
       Reviewable.where(created_by_id: user.id).delete_all
 
+      category_topic_ids = Category.all.pluck(:topic_id).compact
```

Comment:
```suggestion
      category_topic_ids = Category.where("topic_id IS NOT NULL").pluck(:topic_id)
```

Reply:
Not sure if we need this here, see https://github.com/discourse/discourse/pull/11551/files#r547092590